cli/src/action/keygen.rs

```diff
@@ -38,6 +39,29 @@ const DEFAULT_SYSTEM_KEY_NAME: &str = "splinterd";
 
 pub struct KeyGenAction;
 
+#[derive(Debug)]
+pub enum GroupOptions {
+    Auto,
+    Named(String),
+    GroupID(u32),
+}
+
+impl FromStr for GroupOptions {
+    type Err = CliError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "auto" => Ok(GroupOptions::Auto),
```

Comment:
If this is only for parsing the clap args, you should just parse this value there.  See the recent change with ScabbardState

Reply:
Fixed